src/apps/eos/actions/__init__.py

```diff
@@ -0,0 +1,115 @@
+from trezor.crypto.hashlib import sha256
+from trezor.messages.EosTxActionRequest import EosTxActionRequest
+from trezor.messages.MessageType import EosTxActionAck
+from trezor.utils import HashWriter
+
+from apps.eos import helpers, writers
+from apps.eos.actions import layout
+
+
+async def process_action(ctx, sha, action):
+    name = helpers.eos_name_to_string(action.common.name)
+    account = helpers.eos_name_to_string(action.common.account)
+
+    if not check_action(action, name, account):
+        raise ValueError("Unknown action")
+
+    w = bytearray()
+    writers.write_action_common(sha, action.common)
+    if name == "buyram":
```

Comment:
given that "transfer" is disallowed on "eosio", the "eosio" branch should go first and the "transfer" branch second

Reply:
@matejcik It seems redundant, as we already do all possible checks in check_action, here we just sort out correct processor for the action. Can we leave it as is, if we take into account that we already covered all scenarios in check action?

Reply:
No, the requested change fixes functionality.
The following message will pass `check_action` but fail here:
```
{
	account: "mytoken",
	action: "buyram",
	buyram: null,
	unknown: 0xABCDEF,
}
```
I was under the impression that "mytoken.buyram" should be handled as an unknown action with arbitrary data, no?
Otherwise we would force "mytoken" implementers to keep the same arguments as "eosio".

Reply:
Added additional check for account equal "eosio"